### PR TITLE
feat: added activity model

### PR DESCRIPTION
**Pull request by MrSurge87**
## Checklist before requesting a review

- [X] I have performed a self-review of my code
- [ ] If there are changes to custom resolvers, I have added / updated automated tests
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings

### CODE REVIEW

File: models/activity.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      db: { map: "my_start_timestamp", updatedAt: true },
+      validation: { isRequired: true },
+      isIndexed: "unique",
+    }),
+    endTime: timestamp({
+      defaultValue: "2020-10-05T00:00:00-07:00",
+      db: { map: "my_end_timestamp", updatedAt: true },
+      isIndexed: "unique",
+    }),
+  },
```
Comment (dani9401) [inline models/activity.ts:57]: You should include a relationship to the User.ts file here. You can look at other backend PRs for models to see how this is done.  Or you can look at the Keystone 6 documentation on relationships.

File: models/activity.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    },
+  },
+  fields: {
+    name: text({
+      defaultValue: "...",
+      db: { map: "my_text", nativeType: "VarChar(40)" },
+      validation: { isRequired: true },
+      isIndexed: "unique",
+      ui: { displayMode: "textarea" },
+    }),
```
Comment (dani9401) [inline models/activity.ts]: I am actually not sure if you need all these db: references on something simple like a text or integer field. This isn't how I've seen models setup in keystone/apollo but that doesn't necessarily mean it's wrong. Just a thought on this.

File: models/activity.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      delete: ({ session }) => !!session,
+    },
+    item: {
+      update: ({ session, item }) => item.userId === session.data.id,
+      delete: ({ session, item }) => item.userId === session.data.id,
+    },
+  },
+  fields: {
+    name: text({
+      defaultValue: "...",
```
Comment (dani9401) [inline models/activity.ts]: This defaultValue looks weird in Keystone when it loads (see screenshot). I recommend sticking to figma on this and leaving it blank. Then when we build the form on the frontend we can input those placeholder values shown in figma. 

image

File: models/activity.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      isIndexed: "unique",
+      ui: { displayMode: "textarea" },
+    }),
+    amount: integer({
+      defaultValue: 0,
+      db: { map: "my_integer" },
+      isIndexed: "unique",
+    }),
+    unitOfMeasure: text({
+      defaultValue: "...",
```
Comment (dani9401) [inline models/activity.ts]: Same as above.

**Review by dani9401 — CHANGES_REQUESTED** [review]
Hey there! You will need to add this new activities.ts file to models-->index.ts in order to look at it inside Keystone. I left a few other notes/suggestions/plain comments below too.

File: models/activity.ts (hunk 1 above)
Comment (MrSurge87) [reply models/activity.ts:57]: I updated the user relationship.

File: models/activity.ts (hunk 2 above)
Comment (MrSurge87) [reply models/activity.ts]: I looked at other PRs and made mine match more so to them. I did look up the keystone fields and the examples they gave all had those references which is why I used them but the code looks cleaner now.

File: models/activity.ts (hunk 3 above)
Comment (MrSurge87) [reply models/activity.ts]: I made the default values blank now.

File: models/activity.ts (hunk 4 above)
Comment (MrSurge87) [reply models/activity.ts]: I made the default value blank now.

**Comment by dani9401** [comment]
@MrSurge87 Hey! I believe there are new errors on this now. Try running npm run dev to check errors to check them out in your terminal.

**Comment by dani9401** [comment]
@MrSurge87 This is so close and looks great! Much improved, so really good job. 
The only thing left I can see is that it looks like you have all fields required in the model instead of just Name and Start Time to follow the Jira ticket. 
I thought for sure you had that right on a previous review? But now I can't be sure. That should be updated to match the Jira ticket though.

**Comment by MrSurge87** [comment]
> @MrSurge87 This is so close and looks great! Much improved, so really good job. The only thing left I can see is that it looks like you have all fields required in the model instead of just Name and Start Time to follow the Jira ticket. I thought for sure you had that right on a previous review? But now I can't be sure. That should be updated to match the Jira ticket though.

Hi @dani9401 This should be good to go now.

**Review by jahorwitz — APPROVED** [review]

**Pull request merged**